File: com-223 Data Structures/DS/Lab06/p02/Circle.cpp
#include <sstream>
#include "Circle.h"

using namespace std;

Circle::Circle(int aX, int aY, int aR)
: AbstractFigure(aX, aY), r(aR)
{
}

string Circle::toString() const
{
    ostringstream sout;
    sout << "Circle[" << x << ", " << y << ", " << r << "]";
    
    return sout.str();
}


bool Circle::contains(int aX, int aY) const
{
    int dx = x - aX;
    int dy = y - aY;
    return dx * dx + dy * dy <= r * r;
}

File: com-117 Programming2/Programming02/Lab06Problem01/src/Circle.java
import java.awt.*;

public class Circle extends Figure {

	private int r;

	public Circle(int x, int y, int r) {
		super(x, y);
		this.r = r;
	}

	@Override
	public void draw(Graphics g) {
		g.setColor(Color.YELLOW);
		g.fillOval(x - r, y - r, r * 2, r * 2);
	}

	@Override
	public boolean contains(int x, int y) {
		int dx = this.x - x;
		int dy = this.y - y;
		return dx * dx + dy * dy <= r * r;
	}

	@Override
	public String toString() {
		return String.format("Circle: x=%d, y=%d, r=%d", x, y, r);
	}

}
